src/js/angular/clustermanagement/services/cluster-context.service.js

```diff
@@ -0,0 +1,308 @@
+import {cloneDeep} from "lodash";
+import {ClusterViewModel} from "../../models/clustermanagement/cluster";
+
+angular
+    .module('graphdb.framework.clustermanagement.services.cluster-context', [])
+    .factory('ClusterContextService', ClusterContextService);
+
+ClusterContextService.$inject = ['EventEmitterService'];
+
+function ClusterContextService(EventEmitterService) {
```

Comment:
Please, verify that the state kept in this service should not be reset at some point, e.g. on page leaving, dialog closing, etc.

Reply:
done - moved all the logic in the directive